### PR TITLE
NODE-2433/lift-restriction-auth-source

### PR DESCRIPTION
**Pull request by reggi**

### CODE REVIEW

File: test/spec/auth/auth.rst [hunk 1]
```diff
@@ -0,0 +1,1218 @@
+.. role:: javascript(code)
```
Comment (mbroadst) [inline test/spec/auth/auth.rst:1]: I'd prefer not to merge in the whole spec if possible, seems like needless text duplication across all of our driver repos

File: test/spec/connection-string/connection-string-spec.rst [hunk 2]
```diff
@@ -0,0 +1,392 @@
+.. role:: javascript(code)
```
Comment (mbroadst) [inline test/spec/connection-string/connection-string-spec.rst:1]: same with this one

File: test/spec/uri-options/uri-options.rst [hunk 3]
```diff
@@ -0,0 +1,393 @@
+=========================
```
Comment (mbroadst) [inline test/spec/uri-options/uri-options.rst:1]: and this one

File: test/spec/auth/connection-string.yml [hunk 4]
```diff
@@ -287,6 +287,11 @@ tests:
         uri: "mongodb://localhost/foo"
         valid: true
         credential: ~
+    -
```
Comment (mbroadst) [inline test/spec/auth/connection-string.yml:290]: was this the only test added as part of this work?

Comment (reggi) [reply test/spec/auth/connection-string.yml:290]: I cherry-picked this spec, this is the only change. I removed the `.rst` files.

**Pull request merged**